cmd/common/image/inspect.go

```diff
@@ -40,47 +38,37 @@ import (
 func NewInspectCmd(runtime runtime.Runtime) *cobra.Command {
 	var outputMode string
 
+	opGlobalParams := make(map[string]*params.Params)
+
 	outputModes := []string{utils.OutputModeYAML, utils.OutputModeJSON, utils.OutputModeJSONPretty, utils.OutputModeCustom}
 
 	cmd := &cobra.Command{
 		Use:          "inspect",
-		Short:        "Inspect the local gadget image",
+		Short:        "Inspect a gadget image",
 		SilenceUsage: true,
 		Args:         cobra.ExactArgs(1),
 	}
 
 	cmd.PersistentFlags().String("extra-info", "", "Custom info type to display")
 
-	cmd.RunE = func(cmd *cobra.Command, args []string) error {
-		image, err := oci.GetGadgetImageDesc(context.TODO(), args[0])
-		if err != nil {
-			return fmt.Errorf("inspecting image: %w", err)
-		}
+	ociParams := apihelpers.ToParamDescs(ocihandler.OciHandler.InstanceParams()).ToParams()
 
-		runtimeGlobalParams := runtime.GlobalParamDescs().ToParams()
-		runtimeParams := runtime.ParamDescs().ToParams()
-		ociParams := apihelpers.ToParamDescs(ocihandler.OciHandler.InstanceParams()).ToParams()
-		ociParams.Set("pull", oci.PullImageNever)
+	for _, op := range operators.GetDataOperators() {
+		opGlobalParams[op.Name()] = apihelpers.ToParamDescs(op.GlobalParams()).ToParams()
+	}
 
-		// Add operator global flags
-		opGlobalParams := make(map[string]*params.Params)
-		for _, op := range operators.GetDataOperators() {
-			opGlobalParams[op.Name()] = apihelpers.ToParamDescs(op.GlobalParams()).ToParams()
-		}
-		if ociParams, exists := opGlobalParams["oci"]; exists {
-			ociParams.Set("verify-image", "false")
-		}
+	runtimeGlobalParams := runtime.GlobalParamDescs().ToParams()
+	runtimeParams := runtime.ParamDescs().ToParams()
 
-		var info *api.GadgetInfo
-		err = runtime.Init(runtimeGlobalParams)
-		if err != nil {
-			return fmt.Errorf("initializing runtime: %w", err)
-		}
-		defer runtime.Close()
+	runtime.Init(runtimeGlobalParams)
```

Comment:
Calling `runtime.Init(runtimeGlobalParams)` in `NewInspectCmd()` means it will always be executed, even when the user does not run the `image inspect` command. It should be moved below in the `cmd.RunE` callback, so it is only executed when the user uses the `image inspect` command.

This is causing #4301.